### PR TITLE
Increase go version requirement to 1.16

### PR DESCRIPTION
**Pull request by Nelyah**
This is a follow-up to the discussion I had with @deluan [on Discord](https://discord.com/channels/671335427726114836/704303730660737113/819228755918716988).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,11 +9,11 @@ default:
 	go build -ldflags="-X github.com/navidrome/navidrome/consts.gitSha=$(GIT_SHA) -X github.com/navidrome/navidrome/consts.gitTag=master"
 .PHONY: default
 
-dev: check_env
+dev: check_node_env
```
Comment (deluan) [inline Makefile:12]: Should we do the same for node version?

**Review by deluan — COMMENTED** [review]
Hey, thanks for the PR. Was thinking if for `dev` and `server` it would make sense to check for minimum required version, instead of leaving it out or checking for strict version.... What do you think?

File: Makefile (hunk 1 above)
Comment (Nelyah) [reply Makefile:12]: I think this makes total sense. It crossed my mind then I forgot about it.

> Should we do the same for node version? 

We might as well :)

Comment (Nelyah) [reply Makefile]: I've tried to keep it as simple and as portable as I could. The only thing I kind of worry about is `sort -V` (version sort). 

From what I could see (like this [StackOverflow thread (2017)](https://unix.stackexchange.com/questions/350315/how-can-i-sort-a-list-with-major-minor-patch-level-and-sometimes-rc-correctly)), the `-V` flag seems supported by default on OpenBSD and FreeBSD, but not on NetBSD.

**Comment by Nelyah** [comment]
I've made a small fixup to use variables rather than computing twice the installed version number. Let me know if that sounds good to you and I'll rebase.

**Comment by deluan** [comment]
> I've made a small fixup to use variables rather than computing twice the installed version number. Let me know if that sounds good to you and I'll rebase.

Yes, removing duplication is better.

I'll take a look tonight and merge if everything is ok.

**Comment by deluan** [comment]
LGTM. Are you still pushing the change to remove duplication?

**Comment by Nelyah** [comment]
Sorry should probably have commented on that. The last push was to squash the `fixup!` commit (which introduced the addition of variables in the Makefile command).

**Pull request merged**